### PR TITLE
bug 1506073. Lower cpu request for logging when it exceeds limit

### PR DESCRIPTION
**Pull request by jcantrill**
This PR fixes https://bugzilla.redhat.com/show_bug.cgi?id=1506073 by:
* Lowering the CPU request to match the limit when the request is greater then a specified limit

I have an open question on if this is an acceptable change of if it makes the outcome unexpected.  Should we prefer to exit during the deployment and advise the operator to correct their inventory?

### CODE REVIEW

**Comment by jcantrill** [comment]
@portante We are not using facts in this context.  We are using `set_fact` to simply set a variable to be evaluated later

**Comment by portante** [comment]
Why do you have to use `set_fact`?  Can you just put that expression used in the `set_fact` where the fact is referenced?

File: roles/openshift_logging_curator/tasks/main.yaml [hunk 1]
```diff
@@ -90,7 +90,7 @@
     es_host: "{{ openshift_logging_curator_es_host }}"
     es_port: "{{ openshift_logging_curator_es_port }}"
     curator_cpu_limit: "{{ openshift_logging_curator_cpu_limit }}"
-    curator_cpu_request: "{{ openshift_logging_curator_cpu_request }}"
+    curator_cpu_request: "{{ openshift_logging_curator_cpu_request | min_cpu(openshift_logging_curator_cpu_limit | default(none)) }}"
```
Comment (ewolinetz) [inline roles/openshift_logging_curator/tasks/main.yaml]: do we need `| default(none)` for these? is there a reason `min_cpu` wouldn't return a value? also we didn't have a default before... 

Won't block on it, but it does seem unnecessary

**Pull request merged**